### ISSUE TITLE
Updating data in `getData` while in other page cause infinite loop

### ISSUE THREAD

**Issue by padunk** [dai-shi/waku#849]
Replicate steps
1. npm create waku@latest my-app
2. bun i, bun run dev
3. open localhost:3000
4. update const data.title in getData in `about.tsx`
5. HMR error -> infinite loop

This happens vice-versa, meaning if I'm on the `about page` and try to change the `data.title` in `index.tsx` will result in the same error.

![Screenshot 2024-08-25 at 12 00 09](https://github.com/user-attachments/assets/0b1fdce0-6742-4bee-a06f-b6b49e1906fe)

**Comment by dai-shi** [comment]
@padunk Thanks for reporting!

@sandren Is this similar to what you experienced before?

@Aslemammad I wonder if you have a chance to dig into it, or not?

**Comment by rmarscher** [comment]
I couldn't reproduce this. Is it still an issue?

**Comment by dai-shi** [comment]
Not sure if this is something that happens every time, or occasionally.

@padunk Can you confirm with the latest version please?

**Comment by padunk** [comment]
I can confirm that this is still happening in latest version.
```json
    "react": "19.0.0-rc-d6cb4e77-20240911",
    "react-dom": "19.0.0-rc-d6cb4e77-20240911",
    "react-server-dom-webpack": "19.0.0-rc-d6cb4e77-20240911",
    "waku": "0.21.2"
```

Can I help? I need a guide to where to look at this :)

**Comment by dai-shi** [comment]
Thanks for confirming. Yeah, I would like someone to help.
Is this DEV-only issue, right? (in that case rscHmrPlugin is something, but not sure if it's only the place to look at.)

**Comment by padunk** [comment]
Looks like it was fixed in 0.21.3, closing this.
